src/main/java/chess/domain/piece/Knight.java

```diff
@@ -0,0 +1,21 @@
+package chess.domain.piece;
+
+import chess.domain.square.Square;
+
+public class Knight extends Piece {
+
+    private static final int SIDE_STEP = 1;
+    private static final int STRAIGHT_STEP = 2;
+
+    public Knight(Team color) {
+        super(PieceType.KNIGHT, color);
+    }
+
+    @Override
+    public boolean canMove(Square source, Square target) {
+        return (source.calculateFileDiff(target.file()) == SIDE_STEP
+                && source.calculateRankDiff(target.rank()) == STRAIGHT_STEP) ||
+                (source.calculateFileDiff(target.file()) == STRAIGHT_STEP
+                        && source.calculateRankDiff(target.rank()) == SIDE_STEP);
```

Comment:
```suggestion
       if (source.calculateFileDiff(target.file()) == SIDE_STEP && source.calculateRankDiff(target.rank()) == STRAIGHT_STEP) {
            return true;
        }
        return source.calculateFileDiff(target.file()) == STRAIGHT_STEP && source.calculateRankDiff(target.rank()) == SIDE_STEP;
```

조건절은 복잡하면 복잡할수록 가독성이 떨어지니 if를 최대한 활용해보는 방법도 있습니다 😄

Reply:
반영하였습니다. 좋은 방법 감사합니다. ㅎㅎ

```java
public boolean isStraightAndDiagonal(final Square target) {
        if (file.calculateDistance(target.file) == ONE_STEP
                && rank.calculateDistance(target.rank) == TWO_STEP) {
            return true;
        }
        return file.calculateDistance(target.file) == TWO_STEP
                && rank.calculateDistance(target.rank) == ONE_STEP;
    }
```